src/sentry/tasks/split_discover_query_dataset.py

```diff
@@ -0,0 +1,172 @@
+import logging
+from collections.abc import MutableMapping
+from time import sleep, time
+
+import sentry_sdk
+from cachetools import LRUCache
+from celery.exceptions import SoftTimeLimitExceeded
+
+from sentry import options
+from sentry.discover.dataset_split import get_and_save_split_decision_for_query
+from sentry.discover.models import DatasetSourcesTypes, DiscoverSavedQuery, DiscoverSavedQueryTypes
+from sentry.tasks.base import instrumented_task
+from sentry.utils import metrics, snuba
+from sentry.utils.query import RangeQuerySetWrapper
+
+logger = logging.getLogger("sentry.tasks.split_discover_query_dataset")
+
+
+SLEEP_FOR = 5 * 60  # 5 minutes
+MAX_NOOP_ATTEMPTS = 10
+RATE_LIMIT_CACHE: MutableMapping[str | int, int] = LRUCache(maxsize=1000)
```

Comment:
Do we have a rough estimate on the runtime per org yet? (could check snuba admin if not)

Reply:
Ran a redash query actually, we don't have enough distinct org ids with discover saved queries that we need cache with an upper limit - I just made it a regular dict